LISA/Executor.cpp

```diff
@@ -471,11 +539,79 @@ void Executor::doUninstall(std::string type, std::string id, std::string version
         Filesystem::removeDirectory(appStoragePath);
     }
 
-    // TODO invoke maintenace and cleanup
+    doMaintanace();
 
     INFO("finished");
 }
 
+void Executor::doMaintanace()
+{
+    try {
+        // clear tmp
+        Filesystem::removeDirectory(config.getAppsTmpPath());
+        Filesystem::createDirectory(config.getAppsTmpPath());
+
+        // remove installed apps data not present in installed_apps
+        auto appsPathRoot = config.getAppsPath() + Filesystem::LISA_EPOCH + '/';
+        auto foundApps = scanDirectories(appsPathRoot);
+        for (const auto& app : foundApps) {
+            INFO(app);
+            if (dataBase->IsAppInstalled(app.type, app.id, app.version)) {
```

Comment:
The logic should be reversed here: if NOT IsAppInstalled then cleanup dirs

A simple install test shows that the app dirs get deleted immediately after installing a DAC app...